provider/common/instance_configurator.go

```diff
@@ -26,10 +26,10 @@ type InstanceConfigurator interface {
 	ConfigureExternalIpAddress(apiPort int) error
 
 	// Open or close ports.
-	ChangePorts(ipAddress string, insert bool, ports []network.PortRange) error
+	ChangePorts(ipAddress string, insert bool, rules []network.IngressRule) error
 
 	// List all opened ports.
-	FindOpenPorts() ([]network.PortRange, error)
+	FindOpenPorts() ([]network.IngressRule, error)
```

Comment:
This also implies that the Ports() method on the Environ and Instance interfaces also needs to get a new name, which I've now done